docs/content/v0.20/architecture.md

```diff
@@ -1,36 +1,433 @@
+illustrated in Figure 3 (some details might be omitted to make the flow easier
+to understand).
+
+<figure>
+  <img src="./placement-controller.png"  alt="Placement Controller">
+  <figcaption align="center">Figure 3 - Placement Controller</figcaption>
+</figure>
+
+At startup, the controller code sets up the dynamic informers, the event
+handler and the work queue as follows:
+
+- lists all API resources
+- Filters out some resources
+- For each resource:
+  - Creates GVK key
+  - Registers Event Handler
+  - Starts Informer
+  - Indexes informer and lister in a map by GVK key
+- Waits for all caches to sync
+- Starts N workers to process work queue
+
+The reflector is started as part of the informer and watches specific
+resources on the WDS API Server; on create/update/delete object events it
+puts a copy of the object into the local cache. The informer invokes the
+event handler. The handler implements the event handling functions
+(AddFunc, UpdateFunc, DeleteFunc)
+
+A typical event flow for a create/update object event will run as
+follows:
+
+1.  Informer invokes the event handler AddFunc or UpdateFunc
+
+2.  The event handler does some filtering (for example, to ignore update
+    events where the object content is not modified) and then creates a
+    key to store a reference to the object in the work queue. The key
+    contains the GVK key used to retrieve a reference to the informer
+    and lister for that kind of object, and a namespace + name key to
+    retrieve the actual object. Storing the key in the work queue is a
+    common pattern in client-go as the object may have changed in the
+    cache (which is kept updated by the reflector) by the time a worker
+    gets a copy from the work queue. Workers should always receive the
+    key and use it to retrieve the object from the cache.
+
+3.  A worker pulls a key from the work queue, and then does the
+    following processing:
+
+    -  Uses the GVK key to get the reference to the lister for the
+        object
+    -  Gets the object from the lister cache using the NamespacedName of 
+       the object.
+    -  Gets the lister for Placement objects and list all placements
+    -  Iterates on all placements, and for each placement:
+        - evaluates whether the object matches the downsync selection 
+          criteria in the Placement.
+        - If there is a match, list ManagedClusters and
+          find the matching clusters using the label selector expression
+          for clusters.
+        - If there are matching clusters, add the names of the cluster
+          to a hashmap setting the name of the cluster as a key. Cluster
+          groups from different placements are merged together.
+        - If the one of the matched Placements has ` WantSingletonReportedState`
+          set to true, clusters are sorted in alphanumerical order and
+          only the first cluster is selected for delivery.  
```

Comment:
Is anything else done to get the status to come back to WorkStatus objects, or does merely installing the status addon cause status to be returned to WorkStatus objects for all downsynced objects? The reader should be told.

Reply:
At this time, the latter. I have added that in the text.